core/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilders.java

```diff
@@ -89,8 +91,8 @@ private AggregationBuilders() {
     /**
      * Create a new {@link ValueCount} aggregation with the given name.
      */
-    public static ValueCountBuilder count(String name) {
-        return new ValueCountBuilder(name);
+    public static ValueCountAggregator.Factory<ValuesSource> count(String name) {
```

Comment:
We should not expose the ValuesSource class in the client API. Maybe this class does not need to be generic at all?